Detectors/Base/src/MatLayerCylSet.cxx

```diff
@@ -157,37 +157,32 @@ void MatLayerCylSet::dumpToTree(const std::string outName) const
 }
 
 //________________________________________________________________________________
-void MatLayerCylSet::writeToFile(std::string outFName, std::string name)
+void MatLayerCylSet::writeToFile(const std::string& outFName)
 {
   /// store to file
 
   TFile outf(outFName.data(), "recreate");
   if (outf.IsZombie()) {
     return;
   }
-  if (name.empty()) {
-    name = "matBud";
-  }
-  outf.WriteObjectAny(this, Class(), name.data());
+  outf.WriteObjectAny(this, Class(), "ccdb_object");
```

Comment:
I was under the impression that this would be automatically done if you uploaded this using CcdbApi (or the command-line tools)? (I guess you use alternative paths to upload?)

Reply:
Yes, but I use `curl -L -F "blob=@file.root" http://alice-ccdb.cern.ch/<URL>/<from>/<to> --cert ~/.globus/usercert.pem --key ~/.globus/userkey.pem -k` which preserves the name of the object in the file.